### DISCUSSION TITLE
why state chart cant use A -- text --> B

### DISCUSSION THREAD

**Discussion by ts8zs** [mermaid-js/mermaid#4457] [Q&A]
flowchart can use A -- text --> B
but statechart writes A --> B :text

**Comment by knsv** [comment]
The flowchart syntax was created by the mermaid chart project. Then we added state diagram and for simplifying for the authors we (I) used existing PlanUML syntax.

Thats the reason, but it does not have to be that way, see this discussion: https://github.com/orgs/mermaid-js/discussions/4579
